### DISCUSSION TITLE
Ability to use pnpm (or yarn)

### DISCUSSION THREAD

**Discussion by archiewood** [evidence-dev/evidence#2389] [General]
It would be useful if evidence was compatible with other package managers, in particular pnpm which would be helpful for maintaining monorepos.

encountered this issue while looking into #1205 

suspect this is to do with shenanigans in the hidden ./evidence/template folder and or the fact that we explicitly call `npx` [in the cli.js](https://github.com/evidence-dev/evidence/blob/4a512d9d620658595ade21ff12b98986469bc6b5/packages/evidence/cli.js#L168)

## pnpm
```
pnpm i
pnpm run dev
```
result

```
> my-evidence-project@0.0.1 dev /Users/archie/Projects/test123
> evidence dev --open /

sh: vite: command not found
```

## yarn
```
yarn
yarn run dev
```
result
```
yarn run v1.22.19
warning package.json: No license field
warning ../package.json: No license field
$ evidence dev --open /
failed to load config from /Users/archie/Projects/test123/.evidence/template/vite.config.js
error when starting dev server:
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@sveltejs/kit' imported from /Users/archie/Projects/test123/.evidence/template/vite.config.js.timestamp-1695358831162-d0d042e2ff13e.mjs
    at new NodeError (node:internal/errors:405:5)
    at packageResolve (node:internal/modules/esm/resolve:780:9)
    at moduleResolve (node:internal/modules/esm/resolve:829:20)
    at defaultResolve (node:internal/modules/esm/resolve:1034:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:375:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:344:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:220:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36)
✨  Done in 1.23s.
```

**Comment by csjh** [comment]
https://pnpm.io/npmrc#shamefully-hoist fixes for `pnpm`, unsure for `yarn` - `yarn install --flat` does some of the work, but I can't figure out how to make it install peers

**Comment by archiewood** [comment]
@csjh that's cool!

I do wonder though if we are architecting wrong / misconfiguring something though. Shouldn't pnpm "just work".

Is there something fundamentally different about evidence, vs let's say SvelteKit that means we need to do this hoisting?

I also run into a similar kind of dependency hell when trying to use stackblitz (which uses turborepo), which is what makes me think this

**Comment by csjh** [comment]
Yeah, we abuse peer deps a bit to minimize the amount of user `package.json` entries

It might be possible to do a lot of re-exporting from the `@evidence-dev/evidence` package to fix this

**Comment by archiewood** [comment]
See https://github.com/evidence-dev/evidence/issues/1205

**Comment by ItsMeBrianD** [comment]
Just as a note; `npx` _shouldn't_ impact things as far as I know, I've been using it in our monorepo for some time to execute commands. `pnpx` seems to work differently and doesn't use the packages installed in the project in my experience.
